### PR TITLE
qemu_guest_agent: Add reboot/halt support for RHEL 8

### PR DESCRIPTION
**Pull request by BohdanMar**
qemu_guest_agent: add reboot/halt support for RHEL 8
ID:2088394

Signed-off-by: Bogdan Marcynkov <bmarcynk@redhat.com>

### CODE REVIEW

**Comment by BohdanMar** [comment]
(1/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (642.68 s)
 (2/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_install.s390-virtio: PASS (32.78 s)
 (3/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_reboot.s390-virtio: PASS (52.77 s)
 (4/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_halt.s390-virtio: PASS (25.63 s)

**Comment by BohdanMar** [comment]
(1/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (503.04 s)
 (2/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_install.s390-virtio: PASS (24.12 s)
 (3/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_reboot.s390-virtio: PASS (41.72 s)
 (4/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_halt.s390-virtio: PASS (19.05 s)

**Comment by BohdanMar** [comment]
RHEL 9.1:
 (1/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (489.98 s)
 (2/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_install.s390-virtio: PASS (28.32 s)
 (3/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_reboot.s390-virtio: PASS (45.27 s)
 (4/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_halt.s390-virtio: PASS (22.24 s)

**Comment by BohdanMar** [comment]
RHEL 8.6:
(1/3) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (927.73 s)
 (2/3) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_reboot.s390-virtio: PASS (59.68 s)
 (3/3) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_halt.s390-virtio: PASS (27.40 s)

**Comment by BohdanMar** [comment]
@fbq815 Could you please review this patch? Thank you

**Comment by fbq815** [comment]
LGTM @PaulYuuu @ngu-niny please also help bogdan review this patch

File: qemu/tests/cfg/qemu_guest_agent.cfg [hunk 1]
```diff
@@ -105,15 +105,21 @@
             Windows:
                 gagent_guest_reboot_pattern = ""
             s390x:
-                gagent_guest_reboot_pattern = "System Reboot"
+                RHEL.9:
+                    gagent_guest_reboot_pattern = "System Reboot"
```
Comment (ngu-niny) [inline qemu/tests/cfg/qemu_guest_agent.cfg:109]: Can we leave the RHEL.9 pattern as the default one for s390x? So if there is a RHEL.10 guest in the future, and if the pattern works, we don't need to add it anymore. The new code likes following:

            s390x:
                    gagent_guest_reboot_pattern = "System Reboot"
                    RHEL.8:
                              gagent_guest_reboot_pattern = "Started Reboot"

This also applies to the shutdown pattern in line 120.

Comment (BohdanMar) [reply qemu/tests/cfg/qemu_guest_agent.cfg]: Sure, here ais the updates, could you please check?

**Comment by BohdanMar** [comment]
After upfate:
 (1/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (513.41 s)
 (2/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_install.s390-virtio: PASS (24.07 s)
 (3/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_reboot.s390-virtio: PASS (42.20 s)
 (4/4) Host_RHEL.m9.u0.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_halt.s390-virtio: PASS (18.80 s)

File: qemu/tests/cfg/qemu_guest_agent.cfg (hunk 1 above)
Comment (ngu-niny) [reply qemu/tests/cfg/qemu_guest_agent.cfg]: LGTM.

**Pull request merged**